### PR TITLE
Lua dataset/v6

### PR DESCRIPTION
**Pull request by victorjulien**
Initial dataset support for lua.

https://redmine.openinfosecfoundation.org/issues/7243

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2120

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23255

**Comment by victorjulien** [comment]
Passed my QA. Ran this PR with SV PR https://github.com/OISF/suricata-verify/pull/2120. Local pipeline 5131, run 627.

File: src/detect-lua-extensions.c [hunk 1]
```diff
@@ -414,6 +415,90 @@ static int LuaSetFlowint(lua_State *luastate)
...
+    // TODO not sure why we need to do this here. Would expect it to be
+    // a registration time thing.
+    lua_newtable(luastate);
+    lua_pushcfunction(luastate, LuaDatasetGC);
+    lua_setfield(luastate, -2, "__gc");
+    lua_setmetatable(luastate, -2);
+    return 1;
+}
+
+static int LuaDatasetAdd(lua_State *luastate)
```
Comment (jasonish) [inline src/detect-lua-extensions.c:462]: Related to this function: https://github.com/OISF/suricata-verify/pull/2120#discussion_r1838468205

Comment (victorjulien) [reply src/detect-lua-extensions.c:462]: addressed in https://github.com/OISF/suricata/pull/12121

**Pull request closed without merging**